avocado/utils/data_structures.py

```diff
@@ -52,6 +53,63 @@ def geometric_mean(values):
     return math.exp(sum([math.log(number) for number in values]) / no_values)
 
 
+def compare_matrices(matrix1, matrix2, treshold=0.05):
+    """
+    Compare 2 matrices nxm and return a matrix nxm with comparison data
+
+    :param matrix1: Reference Matrix with numeric data
+    :param matrix2: Matrix that will be compared
+    :param treshold: Any difference bigger than this percent treshold will be
+                     reported.
+    :retrun: Matrix with the difference in comparison, number of improvements,
+             number of regressions, total number of comparisons.
+    """
+    improvements = 0
+    regressions = 0
+    same = 0
+    new_matrix = []
+
+    for line1, line2 in zip(matrix1, matrix2):
+        new_line = []
+        elements = izip(line1, line2)
+        try:
+            element1, element2 = elements.next()
+        except StopIteration:
```

Comment:
It'd be nice to add some comments so future readers don't try to "optimize" the code. Here I'd say, `# no data in this row`